core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -625,7 +625,15 @@ object UnitActions {
 
         // City States only take miliary units (and GPs for certain civs)
         if (recipient.isCityState()) {
-            if (unit.isGreatPerson()) return null    // Unless Sweden
+            if (unit.isGreatPerson()) {
+                // Do we have a unique ability to gift GPs?
+                var canGiftGPs = false
+                for (unique in unit.civInfo.getMatchingUniques("Gain [] Influence with a [] gift to a City-State")) {
+                    if(unique.params[1] == "Great Person")
+                        canGiftGPs = true
+                }
+                if (!canGiftGPs) return null
```

Comment:
Can be simplified to 
if(unit.civInfo.getMatchingUniques("Gain [] Influence with a [] gift to a City-State").none{it.params[1]=="Great Person"} return null